src/test/java/com/teragrep/pth_06/planner/walker/conditions/CategoryTableConditionTest.java

```diff
@@ -0,0 +1,228 @@
+                    + "    `filter`         longblob            NOT NULL)";
+            conn.prepareStatement(targetTable).execute();
+        });
+    }
+
+    @AfterAll
+    void tearDown() {
+        Assertions.assertDoesNotThrow(() -> {
+            conn.prepareStatement("DROP ALL OBJECTS").execute(); //h2 clear database
+            conn.close();
+        });
+    }
+
+    @Test
+    void testCondition() {
+        fillTargetTable();
+        DSLContext ctx = DSL.using(conn);
+        Table<?> target1 = ctx
+                .meta()
+                .filterSchemas(s -> s.getName().equals("bloomdb"))
+                .filterTables(t -> !t.getName().equals("filtertype"))
+                .getTables()
+                .get(0);
+        CategoryTableCondition cond = new CategoryTableCondition(target1, 0L);
+        String e = "(\n" + "  bloommatch(\n" + "    (\n" + "      select \"term_0_target\".\"filter\"\n"
+                + "      from \"term_0_target\"\n" + "      where (\n" + "        term_id = 0\n"
+                + "        and type_id = \"bloomdb\".\"target\".\"filter_type_id\"\n" + "      )\n" + "    ),\n"
+                + "    \"bloomdb\".\"target\".\"filter\"\n" + "  ) = true\n"
+                + "  and \"bloomdb\".\"target\".\"filter\" is not null\n" + ")";
+        Assertions.assertEquals(e, cond.condition().toString());
+    }
+
+    @Test
+    void testBloomTermId() {
+        fillTargetTable();
+        DSLContext ctx = DSL.using(conn);
+        Table<?> target1 = ctx
+                .meta()
+                .filterSchemas(s -> s.getName().equals("bloomdb"))
+                .filterTables(t -> !t.getName().equals("filtertype"))
+                .getTables()
+                .get(0);
+        CategoryTableCondition cond = new CategoryTableCondition(target1, 1L);
+        String e = "(\n" + "  bloommatch(\n" + "    (\n" + "      select \"term_1_target\".\"filter\"\n"
+                + "      from \"term_1_target\"\n" + "      where (\n" + "        term_id = 1\n"
+                + "        and type_id = \"bloomdb\".\"target\".\"filter_type_id\"\n" + "      )\n" + "    ),\n"
+                + "    \"bloomdb\".\"target\".\"filter\"\n" + "  ) = true\n"
+                + "  and \"bloomdb\".\"target\".\"filter\" is not null\n" + ")";
+        Assertions.assertEquals(e, cond.condition().toString());
+    }
+
+    @Test
+    public void testEquality() {
+        fillTargetTable();
+        DSLContext ctx = DSL.using(conn);
+        Table<?> target1 = ctx
+                .meta()
+                .filterSchemas(s -> s.getName().equals("bloomdb"))
+                .filterTables(t -> !t.getName().equals("filtertype"))
+                .getTables()
+                .get(0);
+        CategoryTableCondition cond1 = new CategoryTableCondition(target1, 1L);
+        CategoryTableCondition cond2 = new CategoryTableCondition(target1, 1L);
+        Assertions.assertEquals(cond1, cond2);
+        Assertions.assertEquals(cond2, cond1);
```

Comment:
I don't think it is necessary to check cond1==cond2 and then cond2==cond1? spotted this on other tests as well.

Reply:
Cleaned up assertions in equality tests, removed double equality checks and null not equals checks